plugins/timeline/index.js

```diff
@@ -0,0 +1,321 @@
+var q = require('q'),
+    fs = require('fs'),
+    path = require('path'),
+    SauceLabs = require('saucelabs'),
+    https = require('https');
+
+var SAUCE_LOGS_WAIT = 5000;
+
+/**
+ * Outputs information about where your Protractor test is spending its time
+ * to the specified folder. A JSON data file and small index.html to view
+ * it will be created. The page uses Google Charts to show the timeline.
+ *
+ * You enable this plugin in your config file:
+ *
+ *    exports.config = {
+ *      plugins: [{
+ *        path: 'node_modules/protractor/plugins/timelinePlugin.js',
+ *
+ *        // Output json and html will go in this folder.
```

Comment:
Good point. Right now the plugins don't get passed information about the location of the config file, and the way our config parser is set up filenames are made absolute when the configuration is read. This doesn't extend well to plugins - I'll add a TODO that we should rethink this, and make it relative to the current working directory for now.